src/Promitor.Scraper.Host/Validation/MetricDefinitions/ResourceTypes/ContainerInstanceMetricValidator.cs

```diff
@@ -1,18 +1,24 @@
 ﻿using System.Collections.Generic;
+using System.Linq;
 using GuardNet;
+using Promitor.Core.Scraping.Configuration.Model.Metrics;
 using Promitor.Core.Scraping.Configuration.Model.Metrics.ResourceTypes;
+using Promitor.Scraper.Host.Validation.MetricDefinitions.Interfaces;
 
 namespace Promitor.Scraper.Host.Validation.MetricDefinitions.ResourceTypes
 {
-    internal class ContainerInstanceMetricValidator : MetricValidator<ContainerInstanceMetricDefinition>
+    internal class ContainerInstanceMetricValidator : IMetricValidator
     {
-        protected override IEnumerable<string> Validate(ContainerInstanceMetricDefinition containerInstanceMetricDefinition)
+        public IEnumerable<string> Validate(MetricDefinition metricDefinition)
         {
-            Guard.NotNull(containerInstanceMetricDefinition, nameof(containerInstanceMetricDefinition));
+            Guard.NotNull(metricDefinition, nameof(metricDefinition));
 
-            if (string.IsNullOrWhiteSpace(containerInstanceMetricDefinition.ContainerGroup))
+            foreach (var definition in metricDefinition.Resources.Cast<ContainerInstanceResourceDefinition>())
```

Comment:
It'll throw an `InvalidCastException`. We could add a check to make sure all the objects are the right type, and return a validation message if not?

The reason I didn't bother was that I figured this can't happen as a result of a mistake in the config, it would have to happen as a result of a bug in the code being introduced (for example us breaking the AutoMapper mappings). In that case it would just bubble up and give us a stack trace I think.

What's your thoughts?

Reply:
I think you are right, we are good to go here!